src/Compilers/CSharp/Portable/Emitter/Model/ArrayTypeSymbolAdapter.cs

```diff
@@ -39,11 +39,25 @@ IEnumerable<int> Cci.IArrayTypeReference.LowerBounds
         {
             get
             {
-                for (int i = 0; i < this.Rank; ++i)
-                    yield return 0;
+                var lowerBounds = this.LowerBounds;
+
+                if (lowerBounds.IsDefault)
+                {
+                    return DefaultLowerBounds(this.Rank);
+                }
+                else
+                {
+                    return lowerBounds;
+                }
             }
         }
 
+        private static IEnumerable<int> DefaultLowerBounds(int rank)
```

Comment:
This is an option, I guess. However this is not a new code, it is the original implementation of Cci.IArrayTypeReference.LowerBounds extracted into a method. I would prefer to keep decision about changing it outside of scope of this PR. I'll open a separate issue to track this.